#!/usr/bin/env python
# Hazmat configuration

PACKAGE = 'pandora_vision_hazmat'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Example : gen.add("int_param", int_t, 0, "An In.....", 50, 0, 100)
# name type level description def min max


#------------------------Background segmentation----------------------------
gen.add("colorVariance", int_t, 0,"", 30, 0, 500)
gen.add("sideLength", int_t, 0,"", 50, 4, 500)
gen.add("maxAreaThreshold", double_t, 0,"", 3300000, 100000, 8800000)
gen.add("minAreaThreshold", double_t, 0,"", 100, 0, 200000)
gen.add("votingThreshold", double_t, 0,"", 1500, 0, 20000)
gen.add("MOThreshold", double_t, 0,"", 160000, 0, 800000)
gen.add("featureThreshold", int_t, 0,"", 10, 1, 150)
gen.add("numberOfHazmats", int_t, 0,"", 19, 1, 30)

exit(gen.generate(PACKAGE, "hazmat_node", "hazmat_cfg"))
